### PR TITLE
Lint explicit Clone implementations on Copy type

### PR DESCRIPTION
**Pull request by mcarton**
Ref https://github.com/rust-lang/rust/issues/31086.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,7 +6,7 @@ A collection of lints to catch common mistakes and improve your Rust code.
 [Jump to usage instructions](#usage)
 
 ##Lints
-There are 97 lints included in this crate:
+There are 98 lints included in this crate:
```
Comment (Manishearth) [inline README.md]: hype hype hype :100:

File: src/derive.rs [hunk 2]
```diff
@@ -33,66 +35,143 @@ declare_lint! {
     "deriving `Hash` but implementing `PartialEq` explicitly"
 }
 
+/// **What it does:** This lint warns about explicit `Clone` implementation for `Copy` types.
+///
+/// **Why is this bad?** To avoid surprising behaviour, these traits must agree and the behaviour
```
Comment (Manishearth) [inline src/derive.rs]: s/must/should/, s/cannot/should not/, s/override/overridden.

Mention that in almost all situations a `Copy` type should have a `Clone` impl that does nothing more than copy the object, which is what `#[derive(Copy, Clone)]` gets you.

Comment (mcarton) [inline src/derive.rs]: > s/override/overridden

Rhhaa, that’s only the sixth choice for *overrided* in Vim’s spell checker :sweat_smile:

Comment (mcarton) [inline src/derive.rs]: And `Copy` is just a marker trait, so you really cannot change the semantic.

**Comment by llogiq** [comment]
LGTM.

**Comment by bluss** [comment]
Note that if the type is always Copy, an explicit clone impl is much more efficient, something like this one:

```rust
impl<T> Clone for Foo<T> {
    fn clone(&self) -> Self { *self }
}
```

**Pull request merged**